daemon/daemon_unix.go

```diff
@@ -581,15 +560,11 @@ func UsingSystemd(config *config.Config) bool {
 
 // verifyPlatformContainerSettings performs platform-specific validation of the
 // hostconfig and config structures.
-func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.HostConfig, config *containertypes.Config, update bool) ([]string, error) {
-	var warnings []string
+func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.HostConfig, config *containertypes.Config, update bool) (warnings []string, err error) {
 	sysInfo := sysinfo.New(true)
 
 	w, err := verifyContainerResources(&hostConfig.Resources, sysInfo, update)
-
-	// no matter err is nil or not, w could have data in itself.
```

Comment:
AFAIK this comment should not be removed. What it says is "even if err != nil, we still want those warnings". Without such comment, just by looking at the code I want to move the append line to after the err check.

Reply:
Ah, yes, I thought it would be apparent from the function as a whole, but you're right; better to add it back